add note about Expo Go and Analytics (#23509)
expo/expo: docs/pages/guides/using-analytics.mdx

<BoxLink
  title="Aptabase"

>   description={
>     <>
>       Learn how to integrate Aptabase Analytics in your project.
>       <br />
>       <FOOTNOTE tag="span" theme="success">Works with Expo Go</FOOTNOTE>
>     </>
>   }

Reviewer: ```suggestion
  description="Learn how to integrate Aptabase Analytics into your project. Works with Expo Go."
```

I'd rather have this info in the description instead of adding a footnote.

Author: Updated
